src/components/Header/Header.tsx

```diff
@@ -1,20 +1,50 @@
 'use client'
 
-import React, { useState } from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 import styles from './Header.module.sass'
 import Link from 'next/link'
 import CartIcon from '@/svgs/CartIcon'
 import MenuIcon from '@/svgs/MenuIcon'
-import Sidebar from './Sidebar'
+import Sidebar from './subcomponents/Sidebar'
 import NavLinks from '../NavLinks'
+import Cart from './subcomponents/Cart'
 
 const Header = () => {
   const [isMenuOpen, setIsMenuOpen] = useState(false)
+  const [isCartOpen, setIsCartOpen] = useState(false)
 
   const handleMenuClick = () => {
     setIsMenuOpen((prev) => !prev)
   }
 
+  const cartButtonRef = useRef<HTMLButtonElement>(null)
+  const cartRef = useRef<HTMLDivElement>(null)
+
+  const handleCartClick = () => {
+    setIsCartOpen((prev) => !prev)
+  }
+
+  useEffect(() => {
```

Comment:
can you move this to a hook file? since it might be reusable

Reply:
Moved the useEffect hook into a separate hook file.